### PR TITLE
Manage Diacritic in cities names

### PR DESCRIPTION
**Pull request by cvermot**
Manage characters like é à è on cities names... This char should be omited on doctolib URL but are presents on cities names in the json. Exemple : Arès

I didn't find how to be able to use town with - and/or ' char (like Saint-Médard-en-Jalles or Villenave-d'Ornon)

### CODE REVIEW

**Comment by yanndinendal** [comment]
>I didn't find how to be able to use town with - and/or ' char (like Saint-Médard-en-Jalles or Villenave-d'Ornon)

I galèred a bit with this too; here is what I found:

Try a first url (I found the URL used by passing the `-d` param);
something like that: https://www.doctolib.fr/vaccination-covid-19/Saint-M%C3%A9dard-en-Jalles
> Désolé, nous n'avons trouvé aucun professionnel pour votre recherche…

Type the name of thee city in the field and launch the search on the doctolib site;
you will see that the correct URL is https://www.doctolib.fr/vaccination-covid-19/saint-medard-en-jalles, so the normalized city name is `saint-medard-en-jalles`.

**Comment by cvermot** [comment]
Yep, it work with my patch for diactrics with town with "-". Dont know why i didn't manage to work with that before...

But did'nt work with town with ' char (the center name will not been recongnized if we give "villenave-d-ornon" as argument.

File: doctoshotgun.py [hunk 1]
```diff
@@ -196,6 +197,10 @@ def get_patients(self):
         self.master_patient.go()
 
         return self.page.get_patients()
+    
+    def remove_accents(self, string):
+        nfkd = unicodedata.normalize('NFKD', string)
+        return u"".join([c for c in nfkd if not unicodedata.combining(c)])
```
Comment (yanndinendal) [inline doctoshotgun.py]: Works well from what I've tested! :)

Two improvements that could be done as a followup would be to also:
* lowercase the string;
* replace non letters characters (maybe `/\W/`) with a dash (or at least the most common ones: spaces (` `) and apostrophe `'`).

examples:

```py
In [8]: remove_accents("Saint-Médard-en-Jalles")
Out[8]: 'Saint-Medard-en-Jalles'
# should be 'saint-medard-en-jalles' to work

In [9]: remove_accents("Arès")
Out[9]: 'Ares'
# should be 'ares' to work

In [10]: remove_accents("L'Haÿ les roses")
Out[10]: "L'Hay les roses"
# should be 'l-hay-les-roses' to work
```

**Comment by yanndinendal** [comment]
> But did'nt work with town with ' char (the center name will not been recognized if we give "villenave-d-ornon" as argument.

Weird; it seems correct when searching on the website and selecting the autocomplete value for `Villenave-d'Ornon`: https://www.doctolib.fr/vaccination-covid-19/villenave-d-ornon => `villenave-d-ornon` seems correct.

Note that I recommend running the script with the `-d` param, to see if the result is a 404 (wrong city name) or a 503 (Service Temporarily Unavailable). Maybe just try again. :) From my tests, when it's not in 503 it seems to work. :D

File: doctoshotgun.py (hunk 1 above)
Comment (yanndinendal) [inline doctoshotgun.py]: Something like that:

```suggestion
    def normalize(string):
        nfkd = unicodedata.normalize('NFKD', string)
        normalized = u"".join([c for c in nfkd if not unicodedata.combining(c)])
        normalized = re.sub(r'\W', '-', normalized)
        return normalized.lower()
```

File: doctoshotgun.py [hunk 2]
```diff
@@ -376,8 +381,8 @@ def main(self):
             docto.patient = patients[0]
 
         while True:
-            for center in docto.find_centers(args.city):
-                if center['city'].lower() != args.city:
+            for center in docto.find_centers(docto.remove_accents(args.city)):
+                if docto.remove_accents(center['city'].lower()) != docto.remove_accents(args.city):
```
Comment (yanndinendal) [inline doctoshotgun.py]:
```suggestion
                if docto.remove_accents(center['city'].lower()) != docto.normalize(args.city):
```

**Review by yanndinendal — APPROVED** [review]

**Review by yanndinendal — APPROVED** [review]

File: doctoshotgun.py [hunk 3]
```diff
@@ -376,8 +383,8 @@ def main(self):
             docto.patient = patients[0]
 
         while True:
-            for center in docto.find_centers(args.city):
-                if center['city'].lower() != args.city:
+            for center in docto.find_centers(docto.remove_accents(args.city)):
+                if docto.remove_accents(center['city'].lower()) != docto.normalize(args.city):
```
Comment (rbignon) [inline doctoshotgun.py]: You haven't changed the name from `remove_accents` to `normalize` here

Comment (cvermot) [reply doctoshotgun.py]: ooops ! My bad, didn't check !!! Corrected !

Comment (cvermot) [reply doctoshotgun.py]: quick fix on https://github.com/rbignon/doctoshotgun/pull/19/commits/c208be07de845f47f5d6431352a6571007ae1de2

**Comment by snip** [comment]
Seems missing a `@staticmethod` just before `def normalize(string):`

**Comment by rbignon** [comment]
I think you do not test your changes, because there is still a crash :)

**Comment by cvermot** [comment]
OK ! Work now and merged with mutiples cities !

Feel very bad for that ! Excuse me for all the time wasted !

Output now :
> python3 doctoshotgun.py Villenave-d-Ornon,mérignac user@tld.com password
> Available patients are:
> * [0] X Y
> * [1] Z Y
> You want to book a slot for whom patient? 0
> ::: Trying to find a slot in Centre de vaccination Covid-19 - Hôpital Robert Picqué
> ::: Looking for slots in place Hôpital d'Instruction des Armées Robert Picqué - Centre de vaccination
> ::: No availabilities in this center
> ::: Fail, try next center...
> ::: Trying to find a slot in Centre de vaccination COVID-19 Pôle militaire de vaccination - Mérignac - Pin Galant
> ::: Looking for slots in place Pôle militaire de vaccination - Mérignac - Pin Galant
> ::: No availabilities in this center
> ::: Fail, try next center...
> ::: Trying to find a slot in Centre de vaccination Covid-19 - Hôpital Robert Picqué
> ::: Looking for slots in place Hôpital d'Instruction des Armées Robert Picqué - Centre de vaccination
> ::: No availabilities in this center
> ::: Fail, try next center...

**Pull request merged**